data/Building/comment.js

```diff
@@ -65,4 +72,51 @@ async function comment(buildingId, posterId, message) {
   ).exec();
 }
 
-module.exports = comment;
+/**
+ * Modify a comment
+ * @param {string} buildingId The id of the building to take the comment
+ * @param {String} posterId The poster's ID
+ * @param {String} oldMessage The old message to idenify which comment to change
+ * @param {String} newMessage The new message replacing the old message
+ */
+async function modifyComment(buildingId, posterId, oldMessage, newMessage) {
+  ({buildingId, posterId, newMessage} = await validateAndCleanComment(
+      buildingId,
+      posterId,
+      newMessage,
+  ));
+  const commentObj = {
+    posterId,
+    newMessage,
+    timestamp: Date.now(),
+  };
+  return await Building.updateOne(
+      {_id: buildingId},
+      {$set: {'comments.indexOf(oldMessage)': commentObj}},
+  );
+}
+
+/**
+ * Delete a comment
+ * @param {string} buildingId The id of the building to take the comment
+ * @param {String} posterId The poster's ID
+ * @param {String} message The old message to idenify which comment to change
+ */
+async function deleteComment(buildingId, posterId, message) {
```

Comment:
This should be done by message id. If you create a comment on a building, the comment has an _id field.